trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyConfig.java

```diff
@@ -21,19 +21,49 @@
 
 public class TrinoS3ProxyConfig
 {
-    private Optional<String> hostName = Optional.empty();
+    private Optional<String> s3HostName = Optional.empty();
+    private String s3Prefix = "/api/v1/s3Proxy/s3";
+    private String stsPrefix = "/api/v1/s3Proxy/sts";
 
-    @Config("s3proxy.hostname")
-    @ConfigDescription("Hostname to use for REST operations, virtual-host style addressing is only supported if this is set")
-    public TrinoS3ProxyConfig setHostName(String hostName)
+    @Config("s3proxy.s3.hostname")
+    @ConfigDescription("Hostname to use for S3 REST operations, virtual-host style addressing is only supported if this is set")
+    public TrinoS3ProxyConfig setS3HostName(String s3HostName)
     {
-        this.hostName = Optional.ofNullable(hostName);
+        this.s3HostName = Optional.ofNullable(s3HostName);
         return this;
     }
 
     @NotNull
-    public Optional<String> getHostName()
+    public Optional<String> getS3HostName()
     {
-        return hostName;
+        return s3HostName;
+    }
+
+    @Config("s3proxy.s3.prefix")
+    @ConfigDescription("URL Prefix for S3 operations, optional")
+    public TrinoS3ProxyConfig setS3Prefix(String s3Prefix)
```

Comment:
Is `prefix` the right noun? This is the full path.

Reply:
By `prefix` I meant "all S3 operations will be under this path" - meaning all S3 URLs will be this path or something nested within it.

Happy to change it to `s3Path` if you think that's more appropriate but I thought prefix was reasonable

Reply:
I think Path is better